recipe/meta.yaml

```diff
@@ -99,7 +104,13 @@ outputs:
         - {{ pin_subpackage("libclang-cpp", exact=True) }}
         - libcxx {{ cxx_compiler_version }}  # [osx]
         - llvmdev =={{ version }}
-        - zlib                               # [linux or win]
+        # note: these libs may show up as unused in the link check,
+        # but not all the artefacts can be inspected by conda (e.g.
+        # static libs), and they are indeed necessary, given that
```

Comment:
Not true. The build system expects `libz.dylib`, but is unused. They should be added as `ignore_run_exports_from` for the platforms where they are unused.

Reply:
I'm a bit doubtful TBH. Before this, I had followed the link check messages and removed a bunch of host deps, but this just means it'll pick them up from somewhere else (from last [CI](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=675526&view=results) on main):
```
Found ZLIB: /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/lib/libz.dylib (found version "1.2.5") 
Found LibXml2: $PREFIX/lib/libxml2.dylib (found version "2.10.3") 
```

The link check in this PR still says:
```
WARNING (libclang-cpp16): run-exports library package conda-forge::libzlib-1.2.13-h166bdaf_4 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```
but removing it from host leads to build failures. That's because zlib is the only feedstock (I'm aware of) where we separate the actual DSO from the symlink/metadata necessary to use it. But it's a necessary host dep (and it's CMake that will complain if it's not there, not conda).

There's lots of static [libs](https://github.com/regro/libcfgraph/blob/master/artifacts/clangdev/conda-forge/linux-64/clangdev-15.0.7-default_had23c3d_1.json#L1271-L1357) in clangdev and the various outputs. And those static libs can very well contain zlib/zstd etc. in their link interface.

I'll grant that it's a bit of an academic exercise, because since we host-depend on llvmdev (which run-depends on those support libs), they'll _always_ be found in host anyway.

The problem is that for the compiler-infrastructure, where static libs are common, blindly following the link check (which I did previously) can lead down the wrong path.

But I don't mind adding `ignore_run_exports_from` to outputs that only contain shared libs. In the case of `clangdev`, with all its static libs, I think the point stands though (though again they'll be there anyway because here we're also run-depending on llvmdev, which brings them along).

Reply:
I pushed some updates to address this. PTAL